### PR TITLE
LGFS interim fee calc

### PR DESCRIPTION
**Pull request by jsugarman**
#### What
LGFS interim fee calculation

#### Ticket
[CBO-491](https://dsdmoj.atlassian.net/browse/CBO-491)

#### Why
So users do not have to use spreadsheets to workout amounts to be claimed

#### How
Call LAA fee calculator API with relevant query on changing appropriate values on the LGFS interim fee form step page

#### TODO:
- [x] LGFS interim fee front end design implementation (without radios)
- [x] LGFS disbursement front end design implementation 
- [x] LGFS interim fee calc implementation
- [x] LGFS interim fee defendant uplifts **
- [x] rejig and extend specs
- [x] rejig and extend features specs
- [x] refactor

** no misc fees on interim fees, so will have to be dealt with as part of fee calculation (based on actual number of defendants) - PM agrees to use actual number of defendants

#### Further work
- LGFS interim fee type radio implementation - CBO-521
- LGFS interim warrant fee calc implementation - see CBO-518
- LGFS interim fee disbursement only separate form step - CBO-522

### CODE REVIEW

**Review by cleargif — COMMENTED** [review]
👍  Just be careful with the `context` - it can easily pick up multiple dates if it is set to high.

`29/9/1989/1/1/2017`

you will get this back ^^

**Review by cleargif — COMMENTED** [review]
👍

File: app/assets/stylesheets/v1/moj/_HOTFIX_KILLME.scss [hunk 1]
```diff
@@ -54,9 +54,13 @@
 .column-half, .column-two-thirds, .column-quarter, .column-three-quarters, .column-third {
   padding: 0;
 }
+
+// TODO: remove if unused
```
Comment (jsugarman) [inline app/assets/stylesheets/v1/moj/_HOTFIX_KILLME.scss]: remove

File: app/services/claims/fee_calculator/bill_scenario.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      def call
+        bill_scenario
+      end
+
+      private
+
+      def bill_scenario
+        return transfer_detail.bill_scenario if transfer?
+        return find_by_code(fee_type.unique_code) if interim?
+        find_by_code(case_type.fee_type_code) if final?
```
Comment (jsugarman) [inline app/services/claims/fee_calculator/bill_scenario.rb]: remove `if final?`?

File: app/services/claims/fee_calculator/graduated_price.rb [hunk 3]
```diff
@@ -29,10 +29,13 @@ def amount
...
 
-          # In CCCD claiming for defendant uplift is handled as a misc fee.
-          # Defendant uplifts need proper handling as part of the grad fee.
-          # rely on fee calc default now (which assumes one)
-          # options[:number_of_defendants] = 1
+          # For LGFS graduated fee, transfer fee and interim fee types
+          # The number of defendants should be taken from the actual number of
+          # defendants and the misc fee defendant uplift removed
+          #
+          # TODO: retrospectively use actual number of defendants
```
Comment (jsugarman) [inline app/services/claims/fee_calculator/graduated_price.rb:32]: amend

File: spec/services/claims/fee_calculator/calculate_price_spec.rb [hunk 4]
```diff
@@ -37,6 +37,7 @@
   it { is_expected.to delegate_method(:agfs_reform?).to(:claim) }
   it { is_expected.to delegate_method(:case_type).to(:claim) }
   it { is_expected.to delegate_method(:offence).to(:claim) }
+  it { is_expected.to delegate_method(:defendants).to(:claim) }
```
Comment (jsugarman) [inline spec/services/claims/fee_calculator/calculate_price_spec.rb:41]: spec `interim?` delegation too - required by graduated price subclass

**Review by colinbruce — APPROVED** [review]
LGTM :+1:

**Pull request merged**